packages/generate/src/syntax/toEdgeQL.ts

```diff
@@ -1299,6 +1301,7 @@ function shapeToEdgeQL(
     if (val.__kind__ === ExpressionKind.PolyShapeElement) {
       polyType = val.__polyType__;
       val = val.__shapeElement__;
+      hasPolyEl = true;
```

Comment:
Ah, we don't actually have a runtime representation of the type-level "opt in" for this behavior, so this ends up still being an arguably breaking change since we will be returning more. Definitely arguable as to whether or not returning strictly more data is breaking, but anyone who currently has tests for the existing behavior will get failures.

I wonder if we can make a runtime configuration, and actually _infer_ the type-level config from that runtime config?

Reply:
I probably wouldn't say that adding an additional field is a breaking change. User tests may not break. Only if they are asserting exactly or using snapshots.

But I can see how it's not black/white. IMO none of this stuff is. kinda like TS changes lol

Reply:
> User tests may not break. Only if they are asserting exactly or using snapshots.

I think this is _pretty_ common: make a query (or use a data layer more likely) and assert that the returned data is precisely these objects. I agree in general that if you are introducing new properties/data in your own application, maybe you don't consider that a breaking change, but this might warrant a little more care to avoid tripping up people when upgrading point releases.

Reply:
Are you looking for a CLI flag for this as well?

Reply:
Yeah, I think we should add it, like `--enable-polymorphism-as-discriminated-unions` as silly as that sounds. This is non-trivial, but I think it's worth it to avoid going with a major version bump here. I'm honestly open to the major version bump, but that will absolutely delay getting this released.

If you think you have the bandwidth to add the CLI flag, let me know. Otherwise, I'll stick it in the queue and try to get to it (or maybe have @diksipav pick this up) ASAP.

Reply:
```
--future-polymorphism-as-discriminated-unions
```

Reply:
yeah, that was my first thought, but then I thought maybe in the next major we could allow you to still opt-into the old behavior, but now I'm thinking there is no advantage to keeping the old behavior around in the next major.

Reply:
Yeah I don't see a reason to keep the old behavior long term.